internal/server/server.go

```diff
@@ -32,3 +41,76 @@ func New(logger *log.Logger, client Client, templates map[string]*template.Templ
 
 	return http.StripPrefix(prefix, mux)
 }
+
+type RedirectError string
+
+func (e RedirectError) Error() string {
+	return "redirect to " + string(e)
+}
+
+func (e RedirectError) To() string {
+	return string(e)
+}
+
+type StatusError int
+
+func (e StatusError) Error() string {
+	code := e.Code()
+
+	return fmt.Sprintf("%d %s", code, http.StatusText(code))
+}
+
+func (e StatusError) Code() int {
+	return int(e)
+}
+
+type Handler func(w http.ResponseWriter, r *http.Request) error
+
+func errorHandler(name string, logger *log.Logger, tmplError Template, prefix, siriusURL string, next Handler) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if err := next(w, r); err != nil {
+			type v struct {
```

Comment:
`v`! This may be too short even for Go. Also should be pulled up to the top-level instead of hiding in here